src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -620,5 +617,200 @@ public static IDisposable WriteCommonMethod(this CodeWriter writer, MethodSignat
 
             return scope;
         }
+
+        public static CodeWriter WriteBody(this CodeWriter writer, MethodBody methodBody)
+        {
+            foreach (var block in methodBody.Blocks)
+            {
+                WriteBodyBlock(writer, block);
+            }
+
+            return writer;
+        }
+
+        private static void WriteBodyBlock(CodeWriter writer, MethodBodyBlock bodyBlock)
+        {
+            switch (bodyBlock)
+            {
+                case ParameterValidationBlock parameterValidation:
+                    writer.WriteParametersValidation(parameterValidation.Parameters);
+                    break;
+                case DiagnosticScopeMethodBodyBlock diagnosticScope:
+                    using (writer.WriteDiagnosticScope(diagnosticScope.Diagnostic, diagnosticScope.ClientDiagnosticsReference))
+                    {
+                        WriteBodyBlock(writer, diagnosticScope.InnerBlock);
+                    }
+                    break;
+                case MethodBodyLines lines:
+                    foreach (var line in lines.MethodBodySingleLine)
+                    {
+                        writer.Line(line);
+                    }
+                    break;
+            }
+        }
+
+        public static CodeWriter Line(this CodeWriter writer, MethodBodyLine line)
```

Comment:
There already is function Line to write a normal line. It is better to give out a more clear name such as WriteBodyLine.